llama_hub/tools/metaphor/base.py

```diff
@@ -0,0 +1,104 @@
+"""Metaphor tool spec."""
+
+from llama_index.tools.tool_spec.base import BaseToolSpec
+from typing import Optional, List
+from llama_index.readers.schema.base import Document
+import requests
+import urllib.parse
+import datetime
+
+
+class MetaphorToolSpec(BaseToolSpec):
+    """Metaphor tool spec."""
+
+    spec_functions = [
+        "search",
+        "retrieve_documents",
+        "current_date",
+        "find_similar",
+    ]
+
+    def __init__(self, api_key: str) -> None:
+        """Initialize with parameters."""
+        from metaphor_python import Metaphor
+
+        self.client = Metaphor(api_key=api_key)
+
+    def search(
+        self,
+        query: str,
+        num_results: Optional[int] = 10,
```

Comment:
@ajhofmann from your experiments do you think the LLM can infer a reasonable value? or does it usually leave it blank

Reply:
I think the LLM is reasonable at using this field. For example, I changed the wording of `Can you summarize all of the news published in the last month on superconductors` to `some of the news` and it specified num_results = 5, with `all news` it uses the default of 10.